### ISSUE TITLE
transitionTo has empty routerState

### ISSUE THREAD

**Issue by jonathanglasmeyer** [jmurzy/react-router-native#27]
### Version

| Dependency | Version |
| --- | --- |
| react-router-native | master |
| react-router | 3.0.0-alpha.1 |
| react-native | 0.29.0-rc.2 |
### Expected Behavior

`transitionTo` should work.
### Actual Behavior

It throws because `routerState` is an empty object. This makes the code `routes.length - 2` in `getActiveParentRouteType` fail, because `routes` is undefined (destructured from `routerState`). 

I suspect this has something to do with the recent refactorings?

By putting in some `console.logs` i've found out that `syncRouterState` is called _after_ `transitionTo`, which explains why the `routerState` is still set to its initial value of `{}`.

**Comment by jmurzy** [comment]
I'm unable to reproduce this in the example app. Where do you call `transitionTo`?

Either way, this is something flow should've caught. I will try adding type annotations to the history interface.

🍺

**Comment by jonathanglasmeyer** [comment]
I'll try putting up a repro later. 👍🏻

**Comment by jmurzy** [comment]
Did you get a chance to reproduce this in the example app? If you can help me pinpoint the issue, I'd like to fix this over the weekend.

🍺

**Comment by jonathanglasmeyer** [comment]
Here we go. I fetched the newest master (#374b225), `npm install`ed in root and `examples/Aviato`, gulp "copy"ied the `modules` into the example folder.
When doing this in `routes.js`:

``` diff
- <TabsRoute path="master" component={Master}>
+ <TabsRoute path="master" component={Master} onEnter={redirectToNotifications}>
```

OR

``` diff
<Route
  path="/"
+ onEnter={redirectToNotifications}
  component={component('#F5807B')}
  overlayComponent={tabHeaderComponent('#89B2FD', SECOND_HEADER)}
/>
```

This error occurs:
<img width="377" alt="screen shot 2016-07-09 at 10 24 46" src="https://cloud.githubusercontent.com/assets/3755413/16706935/6402ebd4-45bf-11e6-9787-d54cc8947de1.png">

**Comment by jmurzy** [comment]
Fixed in @f367521298c25106986218bf38568f8f3c48beee. Thanks!

🍺

**Comment by jonathanglasmeyer** [comment]
Woohoo!🎉
